src/modules/data/soundData.c

```diff
@@ -119,9 +143,66 @@ SoundData* lovrSoundDataCreateFromFile(struct Blob* blob, bool decode) {
   return soundData;
 }
 
+size_t lovrSoundDataStreamAppendBlob(SoundData *dest, struct Blob* blob) {
+  lovrAssert(dest->ring, "Data can only be appended to a SoundData stream");
+
+  void *store;
+  size_t blobOffset = 0;
+  size_t bytesPerFrame = bytesPerAudioFrame(dest->channels, dest->format);
+  size_t frameCount = blob->size / bytesPerFrame;
+  size_t framesAppended = 0;
+  while(frameCount > 0) {
+    uint32_t availableFrames = frameCount;
+    ma_result acquire_status = ma_pcm_rb_acquire_write(dest->ring, &availableFrames, &store);
+    lovrAssert(acquire_status == MA_SUCCESS, "Failed to acquire ring buffer");
+    memcpy(store, blob->data + blobOffset, availableFrames * bytesPerFrame);
+    ma_result commit_status = ma_pcm_rb_commit_write(dest->ring, availableFrames, store);
+    lovrAssert(commit_status == MA_SUCCESS, "Failed to commit to ring buffer");
+    if (availableFrames == 0) {
+      lovrLog(LOG_WARN, "audio", "SoundData's stream ring buffer is overrun; appended %d and dropping %d frames of data", framesAppended, frameCount);
+      return framesAppended;
+    }
+    
+    frameCount -= availableFrames;
+    blobOffset += availableFrames * bytesPerFrame;
+    framesAppended += availableFrames;
+  }
+  return framesAppended;
+}
+
+size_t lovrSoundDataStreamAppendSound(SoundData *dest, SoundData *src) {
+  lovrAssert(dest->channels == src->channels && dest->sampleRate == src->sampleRate && dest->format == src->format, "Source and destination SoundData formats must match");
+  lovrAssert(src->blob && src->read == lovrSoundDataReadRaw, "Source SoundData must have static PCM data and not be a stream");
+  return lovrSoundDataStreamAppendBlob(dest, src->blob);
+}
+
+void lovrSoundDataSetSample(SoundData* soundData, size_t index, float value) {
+  size_t byteIndex = index * bytesPerAudioFrame(soundData->channels, soundData->format);
+  lovrAssert(byteIndex < soundData->blob->size, "Sample index out of range");
```

Comment:
Any reason not to do `index < soundData->frames` here?

Reply:
No reason. Fixed. Also made sure that soundData has blob :sweat_smile: